### ISSUE TITLE
Since Update to OC 10.13 crons have infinite running sql-queries

### ISSUE THREAD

**Issue by ETES-Stuttgart** [owncloud/core#40971]
### Steps to reproduce
1. Install Update to OC 10.13
2. run the cron-jobs
3. the Job OCA\Files\BackgroundJob\PreviewCleanupJob is leading to infinite running sql-querys and deadlocks
     as I can see the following SQL-Statement is responsible for the issue:
     
```
select `thumb`.`fileid`, `thumb`.`name`, `user_id`  from `oc_mounts`, `oc_filecache` `thumb` left join `oc_filecache` `file`  on `thumb`.`name` = trim(cast(`file`.`fileid` as CHAR(24))) where `thumb`.`parent` in (select `fileid` from `oc_filecache` where `storage` in (select `numeric_id` from `oc_storages` where `id` like 'home::%' or `id` like 'object::user:%') and `path` = 'thumbnails')   and `oc_mounts`.`storage_id` = `thumb`.`storage`   and `file`.`fileid` is null   and `thumb`.`fileid` > '0'   order by `user_id`, `thumb`.`fileid` limit 10;
```
### Expected behaviour
The job should run a view seconds

### Actual behaviour
The job runs multiple hours and will never stop. Also it is leading to issues for mysql-connections so that php-fpm/apache have full scoreboard

### Server configuration
**OS**: RHEL7/8/9

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.4

**ownCloud version:** 10.13

**Updated from an older ownCloud or fresh install:**

**Where did you install ownCloud from:** https://owncloud.com/de/download-server/

I think it could be something like: https://github.com/owncloud/core/issues/40383

We have deactivated the cronjobs right now to stop the issue of infnite running jobs but it would be great if the SQL-Statement would be fixed.

If I investigate the statement I would say the issue is here:
```
MariaDB [owncloud]> select `thumb`.`fileid`, `thumb`.`name`, `user_id`  from `oc_mounts`, `oc_filecache` `thumb`;
ERROR 2008 (HY000): MySQL client run out of memory
```
```
select user_id from oc_mounts;
1237 rows in set (0.094 sec)
```
```
select name from oc_filecache;
355088 rows in set (0.601 sec)
```

Thank you.

**Comment by IljaN** [comment]
We are currently working on a fix. As a workaround you can temporarily remove the offending job from the oc_jobs table to disable execution.

**Comment by ETES-Stuttgart** [comment]
@IljaN thank you for your  reply. 
Kind Regards
Simon

**Comment by cdamken** [comment]
@ETES-Stuttgart Hey Simon, additionally check your DB configuration, It could consume a lot of memory and should write a temp file, but it should not be out of memory as far as I have tested.

**Comment by mihovilkolaric** [comment]
Hi,

I am experiencing the same issue after the upgrade.
@IljaN the offending job in oc_jobs seems to be OCA\Files\BackgroundJob\PreviewCleanupJob - is there any means of "disabling" the job, or should I just remove the entire row? (in case of the latter: do I need to re-add the job manually after the fix, or will it be automatically re-enabled with the next update?)

**Comment by IljaN** [comment]
@mihovilkolaric yes, you can remove the job from the oc_jobs table and re-add it later. Alternatively you can test this PR if you like and have the skill to do so. :)

https://github.com/owncloud/core/pull/40974
https://github.com/owncloud/core/pull/40974.patch

**Comment by mihovilkolaric** [comment]
Hi @IljaN ,
thx a lot - Applying the patch succeeded  (after removing the changes from tests/acceptance/features/cliTrashbin/previewsCleanup.feature - as the acceptance-tests don't seem to be part of the distribution).

First I waited for a few invocations of the cron-job (where nothing happened), before I realized that the this specific background-task is scheduled to be run tomorrow, thus I triggered it manually:
```
$ time ./occ previews:cleanup
...
1302 orphaned previews deleted

real    0m33,050s
user    0m7,297s
sys     0m1,534s
```
followed by another
```
$ time ./occ previews:cleanup
...
1087 orphaned previews deleted

real    0m26,570s
user    0m6,283s
sys     0m1,306s
```

and then also as manual invocation of the backgound-job:
```
$ ./occ background:queue:execute 1057
This command is for maintenance and support purposes. 
This will run the specified background job now. Regular scheduled runs of the job will
continue to happen at their scheduled times. 
If you still want to use this command please confirm the usage by entering: yes
yes
Found job: OCA\Files\BackgroundJob\PreviewCleanupJob with ID 1057
Running job...
Finished in 3 seconds
```

with the log-entry:
`{"reqId":"E8jcE354Qvz5W1JfraUO","level":1,"time":"2023-09-27T19:09:32+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"109 orphaned previews deleted"}`

Any later executions reported "0 orphaned previews deleted", so the cleanup now seems to be finished.
The CPU-Usage during the first two executions (with > 1000 deleted previews) was way lower then before the fix (now: 50% php, 50% mysqld for ~30s - compared to 100% for minutes or hours).

So, from my point of view, the fix works, thanks a lot for the fast response!
